kyo-prelude/shared/src/main/scala/kyo/Parse.scala

```diff
@@ -248,6 +249,70 @@ object Parse:
                 Parse.fail(seq.map(_._1), "Ambiguous parse - multiple results found")
         }
 
+    /** Runs a parser on a stream of text input, emitting parsed results as they become available. This streaming parser accumulates text
+      * chunks and continuously attempts to parse complete results, handling partial inputs and backtracking as needed.
+      *
+      * @param input
+      *   Stream of text chunks to parse
+      * @param v
+      *   Parser to run on the accumulated text
+      * @tparam A
+      *   Type of parsed result
+      * @tparam S
+      *   Effects required by input stream
+      * @tparam S2
+      *   Effects required by parser
+      * @return
+      *   Stream of successfully parsed results, which can abort with ParseFailed
+      */
+    def run[A: Flat, S, S2](input: Stream[Text, S])(v: A < (Parse & S2))(
```

Comment:
This initial implementation isn't very efficient. Instead of always restarting the parser execution from start, it could use a suspension to "pause" the parsing where more input is needed so intermediate parsings aren't discarded. We'll need a custom `ArrowEffect` to optimize it.

Reply:
yeah, makes sense. I can play with it when optimizing the effect.